app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt

```diff
@@ -1,31 +1,155 @@
+    // Main Scaffold layout for ProfileScreen with Bottom Navigation
+    Scaffold(
+        modifier = Modifier.testTag("profileScreen"),
+        bottomBar = {
+            BottomNavigationMenu(
+                onTabSelect = { route -> navigationActions.navigateTo(route) },
+                tabList = LIST_TOP_LEVEL_DESTINATION,
+                selectedItem = navigationActions.currentRoute(),
+            )
+        },
+        content = { paddingValues ->
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(paddingValues)
+                    .testTag("profileScreenContent"),
+                contentAlignment = Alignment.Center
+            ) {
+                if (isSigningOut) {
+                    CircularProgressIndicator(modifier = Modifier.testTag("signingOutIndicator"))
+                } else if (isLoading) {
+                    CircularProgressIndicator(modifier = Modifier.testTag("loadingIndicator"))
+                } else {
+                    user?.let { userData ->
+                        ProfileContent(
+                            userData = userData,
+                            onSignOut = { isSigningOut = true }
+                        )
+                    } ?: run {
+                        Text("No user data available", modifier = Modifier.testTag("noUserData"))
+                    }
+                }
+            }
+        }
+    )
 }
+
+@Composable
+fun ProfileContent(userData: User, onSignOut: () -> Unit) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+            .testTag("profileContent"),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        // Display the profile picture if available
+        if (userData.profilePicture.isNotEmpty()) {
+            Image(
+                painter = rememberAsyncImagePainter(model = userData.profilePicture),
+                contentDescription = "Profile Picture",
+                modifier = Modifier
+                    .size(128.dp)
+                    .clip(CircleShape)
+                    .testTag("profilePicture")
+            )
+        } else {
+            Icon(
+                imageVector = Icons.Default.AccountCircle,
+                contentDescription = "Default Profile Picture",
+                modifier = Modifier
+                    .size(128.dp)
+                    .testTag("defaultProfilePicture")
+            )
+        }
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        // Display user name and email
+        Text(
+            text = "Name: ${userData.name.takeIf { it.isNotEmpty() } ?: "No name available"}",
+            style = MaterialTheme.typography.titleLarge,
+            modifier = Modifier.testTag("userName")
+        )
+        Text(
+            text = "Email: ${userData.email.takeIf { it.isNotEmpty() } ?: "No email available"}",
```

Comment:
The email should always be available

Reply:
I see your point, but since we already have the uid from Firebase, we don’t need to store the account email in the User model. We can always retrieve the account email directly from Firebase whenever it’s needed. This way, we can keep the email as an editable field for the user, allowing them to display a different email on their profile if they want to, while still keeping their login information tied to the Firebase account. It keeps the data cleaner and avoids redundancy.

Reply:
The email should not be editable, because then you need to verify it. So for example someone can impersonate my email without any verification. Having the email as the one from your account verifies intergrity.

Reply:
I understand your point, but I don't fully agree with it. I initially designed the email to be editable for flexibility, allowing users to display a different email on their profile if they wanted to. However, I see the concern about verification and integrity. To avoid any potential issues, I'll go ahead and make the email non-editable and always retrieve it directly from the Firebase account. Thanks for the feedback!